src/java.base/share/classes/jdk/internal/loader/Resource.java

```diff
@@ -87,24 +86,7 @@ public byte[] getBytes() throws IOException {
         // Get stream before content length so that a FileNotFoundException
         // can propagate upwards without being caught too early
         InputStream in = cachedInputStream();
-
-        // This code has been uglified to protect against interrupts.
-        // Even if a thread has been interrupted when loading resources,
-        // the IO should not abort, so must carefully retry, failing only
-        // if the retry leads to some other IO exception.
-
-        boolean isInterrupted = Thread.interrupted();
-        int len;
-        for (;;) {
-            try {
-                len = getContentLength();
-                break;
-            } catch (InterruptedIOException iioe) {
-                Thread.interrupted();
-                isInterrupted = true;
-            }
-        }
-
+        int len = getContentLength();
```

Comment:
Pre-existing question but I wonder if this should move into the try block so that the input stream is closed in the event of getContentLength failing.

Reply:
I agree, that seems like a good idea.